tools/dependencies/pixi_dependency_updater.py

```diff
@@ -0,0 +1,261 @@
+        for os_name, conda_version in conda_versions.items():
+            pixi_version = pixi_versions.get(os_name, "")
+            if not pixi_version:
+                update_pin_in_pixi_manifest(package, os_name, conda_version)
+                made_change_to_pixi = True
+            # if you run pixi add with no spec specified, pixi will add some amount of version restraint;
+            # therefore, we want to check that the conda version has actually been specified before updating the toml
+            if pixi_version and conda_version and not _compare_version_numbers(conda_version, pixi_version):
+                update_pin_in_pixi_manifest(package, os_name, conda_version)
+                made_change_to_pixi = True
+
+    return made_change_to_pixi
+
+
+def compare_conda_and_pixi_envs(conda_env: DependencyPins, pixi_env: DependencyPins) -> Sequence[str]:
+    extra_in_conda = set(conda_env.keys()) - set(pixi_env.keys())
+    extra_in_pixi = set(pixi_env.keys()) - set(conda_env.keys())
+    messages = []
+    if extra_in_conda:
+        messages.append("Packages in the conda environment not found within pixi:\n" + "\n".join(extra_in_conda))
+        return messages
+    if extra_in_pixi:
+        messages.append("Packages in the pixi environment not found within conda:\n" + "\n".join(extra_in_pixi))
+        return messages
+
+    for package, conda_versions in conda_env.items():
+        pixi_versions = pixi_env.get(package, {})
+        for os_name, conda_version in conda_versions.items():
+            pixi_version = pixi_versions.get(os_name, "")
+            if not pixi_version:
+                messages.append(f"{package} found for os: {os_name} in the conda environment but not in the pixi environment")
+            if pixi_version and conda_version and not _compare_version_numbers(conda_version, pixi_version):
+                messages.append(
+                    f"{package}{conda_version} found for os: {os_name} in the conda environment but has version pin {pixi_version}"
+                    f" in the pixi environment"
+                )
+        for os_name, pixi_version in pixi_versions.items():
+            conda_version = conda_versions.get(os_name, None)
+            if conda_version is None:
+                messages.append(f"{package} found for os: {os_name} in the pixi environment but not in the conda environment")
+            if conda_version and not _compare_version_numbers(conda_version, pixi_version):
+                messages.append(
+                    f"{package}{pixi_version} found for os: {os_name} in the pixi environment but has version pin {conda_version}"
+                    f" in the conda environment"
+                )
+
+    return messages
+
+
+def main(argv: Sequence[str] = None) -> int:
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "filenames",
+        nargs="*",
+        help="Filenames pre-commit believes are changed.",
```

Comment:
A `description` or `epilog` argument would be a great place to tell people that this tries to sync dependencies between mantid-developer and pixi.toml.